### DISCUSSION TITLE
Global Store variable getting updated on updating a local variable

### DISCUSSION THREAD

**Discussion by anjanesh** [alpinejs/alpine#3860] [4. General]
I have an **Alpine.store('global_data').aws['storage_partitions']**

I have in my `x-data` in one inner DIV :

```
x-data="{ partitions: [], }"
```

with a form

```
<input type="text" x-model="partition.name" />
<input type="text" x-model="partition.size" />

<button
    type="button"
    @click="
        p = { name: 'Disk ' + (partitions.length + 1), size: 0 };
        partitions.push(p);
        // When p is updated so does the entry in $store.global_data.aws['storage_partitions'] ? Seems to be
        $store.global_data.aws['storage_partitions'].push(p);
    "
>Add Partition</button>
```

But I don't get how an entry in the global store, **Alpine.store('global_data').aws['storage_partitions']** gets updated on updating the local Alpine variable array` partitions` !

**Comment by SimoTod** [comment]
Can you post the full alpine component or a simplified version to reproduce. The issue is likely to be in the part of the code you didn't post

**Comment by ekwoka** [comment]
Where is the x-data?

Also, you don't need to do `['']` around valid key literals.

Just

```js
$store.global_data.aws.storage_partition
```
